--- slock.c	2020-10-18 04:27:44.313408708 +0300
+++ slock.c.new	2020-10-18 04:31:30.043405049 +0300
@@ -156,6 +156,15 @@
 			    IsPFKey(ksym) ||
 			    IsPrivateKeypadKey(ksym))
 				continue;
+			if (ev.xkey.state & ControlMask)
+			{
+				if (ev.xkey.keycode == 44) // Ctrl+J
+					ksym = XK_Return;
+				else if (ev.xkey.keycode == 43) // Ctrl+H
+					ksym = XK_BackSpace;
+				else if (ev.xkey.keycode == 30) // Ctrl+U
+					ksym = XK_Escape;
+			}
 			switch (ksym) {
 			case XK_Return:
 				passwd[len] = '\0';
