diff --git a/x.c b/x.c
index 696ade9..23ed0b8 100644
--- a/x.c
+++ b/x.c
@@ -25,7 +25,7 @@
 /* types used in config.h */
 typedef struct {
 	uint mod;
-	KeySym keysym;
+	KeyCode keycode;
 	void (*func)(const Arg *);
 	const Arg arg;
 } Shortcut;
@@ -2017,6 +2017,7 @@
 {
 	XKeyEvent *e = &ev->xkey;
 	KeySym ksym;
+	KeyCode kcode;
 	char *buf = NULL, *customkey;
 	int len = 0;
 	int buf_size = 64;
@@ -2049,9 +2050,10 @@
 		// so it is not as critical
 		len = XLookupString(e, buf, buf_size, &ksym, NULL);
 	}
+	kcode = e->keycode;
 	/* 1. shortcuts */
 	for (bp = shortcuts; bp < shortcuts + LEN(shortcuts); bp++) {
-		if (ksym == bp->keysym && match(bp->mod, e->state)) {
+		if (kcode == bp->keycode && match(bp->mod, e->state)) {
 			bp->func(&(bp->arg));
 			goto cleanup;
 		}
